### ISSUE TITLE
List of Shells is blank + error in dev tools

### ISSUE THREAD

**Issue by christopherpickering** [Eugeny/terminus#2744]
Version: 1.0.112
Platform: win32 10.0.17763
Plugins: none

Hi,
There are no shells listed in the settings. When I open dev tools this is the error message. Any ideas what I can do to fix? I also cannot create profiles.

```
C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:4302 ERROR Error: Uncaught (in promise): TypeError: Cannot read property 'value' of undefined
TypeError: Cannot read property 'value' of undefined
    at WSLShellProvider.eval (wsl.ts:91)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.js:65)
    at new ZoneAwarePromise (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:913)
    at __awaiter (tslib.es6.js:61)
    at WSLShellProvider.provide (wsl.ts:41)
    at eval (terminal.service.ts:134)
    at Array.map (<anonymous>)
    at TerminalService.eval (terminal.service.ts:134)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.js:65)
    at new ZoneAwarePromise (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:913)
    at __awaiter (tslib.es6.js:61)
    at TerminalService.getShells (terminal.service.ts:133)
    at TerminalService.eval (terminal.service.ts:140)
    at Generator.next (<anonymous>)
    at eval (tslib.es6.js:65)
    at new ZoneAwarePromise (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:913)
    at __awaiter (tslib.es6.js:61)
    at TerminalService.reloadShells (terminal.service.ts:138)
    at new TerminalService (terminal.service.ts:30)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:11534
    at _callFactory (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22110)
    at _createProviderInstance (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22068)
    at resolveNgModuleDep (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22041)
    at _callFactory (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22112)
    at _createProviderInstance (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22068)
    at resolveNgModuleDep (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22026)
    at NgModuleRef_.get (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22647)
    at injectInjectorOnly (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:1032)
    at ɵɵinject (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:1037)
    at injectArgs (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:1129)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:11534
    at _callFactory (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22110)
    at _createProviderInstance (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22068)
    at resolveNgModuleDep (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22041)
    at _createClass (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22101)
    at _createProviderInstance (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22065)
    at initNgModule (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:21995)
    at new NgModuleRef_ (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22635)
    at Object.createNgModuleRef (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:22624)
    at NgModuleFactory_.create (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:32725)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28808
    at ZoneDelegate.invoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:386)
    at Object.onInvoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28264)
    at ZoneDelegate.invoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:385)
    at Zone.run (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:143)
    at NgZone.run (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28159)
    at PlatformRef.bootstrapModuleFactory (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28806)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28855
    at resolvePromise (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:832)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:739
    at rejected (tslib.es6.js:63)
    at ZoneDelegate.invoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:386)
    at Object.onInvoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28264)
    at ZoneDelegate.invoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:385)
    at Zone.run (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:143)
    at C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:891
    at ZoneDelegate.invokeTask (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:421)
    at Object.onInvokeTask (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:28252)
    at ZoneDelegate.invokeTask (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:420)
    at Zone.runTask (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:188)
    at drainMicroTaskQueue (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:601)
    at ZoneTask.invokeTask (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:507)
    at ZoneTask.invoke (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:492)
    at timer (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\zone.js\dist\zone.js:3034)
    at sentryWrapped (C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\dist\sentry.js:1882)
defaultErrorLogger @ C:\Users\chris-pickering\AppData\Local\Programs\Terminus\resources\app.asar\node_modules\@angular\core\bundles\core.umd.js:4302
```
![Capture](https://user-images.githubusercontent.com/17788706/85895359-2b839600-b7bc-11ea-9197-fe8e099421c0.PNG)

Thanks!

**Comment by christopherpickering** [comment]
A few more notes:
- this was a fresh win 10 enterprise install
- installed wsl
- installed terminus
- within 10 mins ( I wasn't paying attention to the order of events as I wasn't expecting a problem) I was able to use terminus w/out problems, and I activated alpine wsl from https://github.com/yuk7/AlpineWSL, then next time I restarted terminus the shell list was blank. I had a saved profile that worked, but once the profile was removed I could not create new profiles.
- cmder still has a full list of prompts (I think terminus crashes because it loads wsl first?)
- I uninstalled terminus, removed user folder from /appdata/roaming (or was it local?) and reinstalled, but no luck
- I changed default wsl from Alpine to a fresh install of ubuntu 20 and restarted PC and still have the same error.

Any ideas on where to look next would be appreciated! Terminus looks like a nice alternative to cmder.
Thanks!

**Comment by Eugeny** [comment]
Please post a dump of your `HKCU\Software\Microsoft\Windows\CurrentVersion\Lxss` registry branch

**Comment by christopherpickering** [comment]
![1E70DA8B-AD9B-4B1B-B11C-119AAED430BA](https://user-images.githubusercontent.com/17788706/85954184-3b75b400-b93b-11ea-8394-ac9c950782a9.png)

![964D402D-8FA2-4519-8DBF-BF5CBD20CB97](https://user-images.githubusercontent.com/17788706/85954187-47617600-b93b-11ea-8ef7-6db2bcb7ac70.png)

![0ECBEE8D-5609-4EBD-BC7D-317AD6C13ADA](https://user-images.githubusercontent.com/17788706/85954193-521c0b00-b93b-11ea-9a7f-d6b8dcd2be0b.png)

**Comment by christopherpickering** [comment]
Hi,
I did a few things that seem to make it work:
1. Disable WSL in windows features, then uninstall Ubuntu & delete Alpine.
2. Delete child registry keys for Lxss & delete "DefaultDistribution" from Lxss.
(restart)
3. Checked Terminus - all remaining shells are listed again (cmd, powershell, git bash, etc)
4. Enabled WSL
(restart)
5. Checked Terminus - all remaining shells are listed again (cmd, powershell, git bash, etc)
6. Installed Ubuntu 20 first this time, added profile in Terminus (default shell set to CMD first)
7. Checked Terminus - all remaining shells are listed again (cmd, powershell, git bash, etc + Ubuntu). I'm able to see and open all
8. Install Alpine...https://github.com/yuk7/AlpineWSL/releases/tag/3.11.5-1

everything is broken again.

It looks like it breaks where it tries to read "flags" form the registry, so I added in a flag key to the Apline reg entry, and left the default value.

![image](https://user-images.githubusercontent.com/17788706/86130854-69c5d180-baaa-11ea-87f8-39854d411c39.png)

After restarting terminus this seems to fix the problem.

![image](https://user-images.githubusercontent.com/17788706/86130917-88c46380-baaa-11ea-8758-79ebfd4eca46.png)
